music_assistant/providers/ard_audiothek/__init__.py

```diff
@@ -0,0 +1,894 @@
+methods should be implemented as async methods. If you are not familiar with
+async programming in Python, we recommend you to read up on it first.
+If you are using a third-party library that is not async, you can need to use the several
+helper methods such as asyncio.to_thread or the create_task in the mass object to wrap
+the calls to the library in a thread.
+
+To add a new provider to Music Assistant, you need to create a new folder
+in the providers folder with the name of your provider (e.g. 'my_music_provider').
+In that folder you should create (at least) a __init__.py file and a manifest.json file.
+
+Optional is an icon.svg file that will be used as the icon for the provider in the UI,
+but we also support that you specify a material design icon in the manifest.json file.
+
+IMPORTANT NOTE:
+We strongly recommend developing on either macOS or Linux and start your development
+environment by running the setup.sh script in the scripts folder of the repository.
+This will create a virtual environment and install all dependencies needed for development.
+See also our general DEVELOPMENT.md guide in the repository for more information.
+
+"""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator, Sequence
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any
+
+from gql import Client
+from gql.transport.aiohttp import AIOHTTPTransport
+from music_assistant_models.config_entries import ConfigEntry
+from music_assistant_models.enums import (
+    ConfigEntryType,
+    ContentType,
+    ImageType,
+    LinkType,
+    MediaType,
+    ProviderFeature,
+    StreamType,
+)
+from music_assistant_models.errors import LoginFailed, MediaNotFoundError
+from music_assistant_models.media_items import (
+    AudioFormat,
+    BrowseFolder,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemLink,
+    MediaItemType,
+    Podcast,
+    PodcastEpisode,
+    ProviderMapping,
+    Radio,
+    SearchResults,
+)
+from music_assistant_models.streamdetails import StreamDetails
+
+from music_assistant.controllers.cache import use_cache
+from music_assistant.models.music_provider import MusicProvider
+from music_assistant.providers.ard_audiothek.database_queries import (
+    get_history_query,
+    get_subscriptions_query,
+    livestream_query,
+    organizations_query,
+    publication_services_query,
+    publications_list_query,
+    search_radios_query,
+    search_shows_query,
+    show_episode_query,
+    show_length_query,
+    show_query,
+    update_history_entry,
+)
+
+if TYPE_CHECKING:
+    from aiohttp import ClientSession
+    from music_assistant_models.config_entries import ConfigValueType, ProviderConfig
+    from music_assistant_models.provider import ProviderManifest
+
+    from music_assistant.mass import MusicAssistant
+    from music_assistant.models import ProviderInstanceType
+
+# Config for login
+CONF_USER_NAME = "username"
```

Comment:
You should be using the MA constants here I think.
https://github.com/music-assistant/server/blob/dev/music_assistant/constants.py